radar-commons-android/src/main/java/org/radarcns/android/auth/portal/RefreshTokenResultReceiver.java

```diff
@@ -0,0 +1,62 @@
+package org.radarcns.android.auth.portal;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.ResultReceiver;
+
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigException;
+
+import org.radarcns.android.auth.AppAuthState;
+import org.radarcns.android.auth.LoginListener;
+import org.radarcns.android.auth.LoginManager;
+import org.radarcns.producer.AuthenticationException;
+
+import java.io.IOException;
+import java.net.ConnectException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.radarcns.android.auth.portal.ManagementPortalService.MANAGEMENT_PORTAL_REFRESH;
+import static org.radarcns.android.auth.portal.ManagementPortalService.REQUEST_FAILED_REASON;
+import static org.radarcns.android.auth.portal.ManagementPortalService.REQUEST_FAILED_REASON_CONFIGURATION;
+import static org.radarcns.android.auth.portal.ManagementPortalService.REQUEST_FAILED_REASON_DISCONNECTED;
+import static org.radarcns.android.auth.portal.ManagementPortalService.REQUEST_FAILED_REASON_IO;
+import static org.radarcns.android.auth.portal.ManagementPortalService.REQUEST_FAILED_REASON_UNAUTHORIZED;
+
+public class RefreshTokenResultReceiver extends ResultReceiver{
```

Comment:
@blootsvoets What do you think? Is this a good idea?

Reply:
I'm not sure if Android likes it... Because the ResultReceiver is `Parcelable` (a form of serializable), but this subclass is not...